### PR TITLE
Week1 Assignments

### PR DESCRIPTION
**Pull request by joeshajan**
These are the assignments in the first week.

The first assignment is about How the browser works and it's in the readme file in the folder Exercise 1.1

The Second assignment is to clone a page and its in folder Exercise 1.2 and open the index.html file to view this.

### CODE REVIEW

**Comment by joeshajan** [comment]
These are the assignments in the first week.

The first assignment is about **How the browser works** and it's in the readme file in the folder **Exercise 1.1**

The Second assignment is to clone a page and its in folder **Exercise 1.2** and open the **index.html** file to view this.

File: Week-1/Exersice 1.2/styles/bottom-bar.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#bottom-bar {
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-1/Exersice 1.2/styles/bottom-bar.css]: @rahulm2 changed id to class

File: Week-1/Exersice 1.2/styles/bottom-bar.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#bottom-bar :last-child>ul {
+    height: 100%;
+    list-style-type: none;
+    margin: 0;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    font-size: 12px;
+}
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-1/Exersice 1.2/styles/bottom-bar.css]: @rahulm2 added this setting in vscode

File: Week-1/Exersice 1.1/readme.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+![High level components](./images/High-level-components.png?raw=true "Title")
+
+# Rendering engine
+
+The responsibility of the rendering engine is to display the requested contents on the browser screen.
+
+The different steps of the rendering engine
+
+1. **Parce HTML and create a dom tree:** The rendering engine will start parsing the HTML document and convert elements to DOM nodes in a tree called the "content tree".
+2. **Create render tree from dom tree:** While the DOM tree is being constructed, the browser constructs another tree, the render tree. This tree is of visual elements in the order in which they will be displayed. It is the visual representation of the document. The purpose of this tree is to enable painting the contents in their correct order.
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-1/Exersice 1.1/readme.md:30]: Added explanation for dom,cssom rendering tree and how it blocks

File: Week-1/Exersice 1.1/readme.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+3. **Layout the render tree:** When the renderer is created and added to the tree, it does not have a position and size. Calculating these values is called layout or reflow. HTML uses a flow-based layout. The layout is a recursive process. It begins at the root renderer, which corresponds to the \<html\> element of the HTML document. The layout continues recursively through some or all of the frame hierarchy.
+4. **Painting:** In the painting stage, the render tree is traversed and the renderer's "paint()" method is called to display content on the screen. The painting uses the UI infrastructure component.
+
+![How rendering works](./images/How-rendering-works.png?raw=true "Title")
+
+# Parsing
+
+Parsing a document means translating it to a structure the code can use. The result of parsing is usually a tree of nodes that represent the structure of the document. This is called a parse tree or a syntax tree.
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-1/Exersice 1.1/readme.md:62]: Added explanation for tokenizer in parsing

**Review by rahulm2 — APPROVED** [review]
lgtm. Nice work

**Pull request merged**